Descarge la encuesta multiproposito 2011 de Bogotá (archivos a.csv, b.csv, c.csv, d.csv, ing_gastos.csv y localidad.csv) y realice los puntos para practicar para el parcial

In [67]:
import os
import pandas as pd
import numpy as np

In [2]:
os.chdir(r'C:\Users\Home\Documents\Laboral2020\Konrad Lorenz\Datos Estructurados y No estructurados\Clase 3\Taller1')

In [3]:
os.listdir()

['a.csv',
 'b.csv',
 'c.csv',
 'd.csv',
 'ing_gastos.csv',
 'localidad.csv',
 'Taller 1 Datos estructurados y no estructurados.docx',
 '~$ller 1 Datos estructurados y no estructurados.docx']

In [26]:
a = pd.read_csv('a.csv')
b = pd.read_csv('b.csv')
c = pd.read_csv('c.csv')
d = pd.read_csv('d.csv')
ing_gastos = pd.read_csv('ing_gastos.csv')
localidad = pd.read_csv('localidad.csv')

C:\Users\Home\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3057: DtypeWarning: Columns (141,142) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [27]:
c.columns

Index(['fex_c', 'directorio_hog', 'directorio', 'hog', 'c1', 'c2_valor', 'c3',
       'c4a', 'c4b', 'c4c',
       ...
       'c50e', 'c50f', 'c50g', 'c50h', 'c50i', 'c50j', 'c50_otro', 'c13j_otro',
       'c8j_otro', 'c12_cual'],
      dtype='object', length=145)

Calcule el promedio que pagan los encuestados por concepto de arriendo (C2, C10_Valor) según tipo de vivienda (B12) y localidad. Nota: sólo tenga en cuenta a los que paguen arriendo. Presente la tabla ordenada por localidad. 

In [42]:
punto0_b = b.filter(['directorio','b12'])
punto0_c = c.filter(['directorio','c1', 'c10_valor'])
punto0=punto0_b.merge(punto0_c, on='directorio', how='inner')
punto0=punto0.merge(a, on='directorio', how='inner')
punto0=punto0.merge(localidad, on='localidad', how='inner')

Los que pagan arriendo c1 = 3

In [56]:
consulta0=punto0.query("c1 == 3").groupby(['nom_loc', 'b12'], as_index=False).agg({'c10_valor':['mean']})

In [57]:
type(consulta0)

pandas.core.frame.DataFrame

In [59]:
consulta0 = consulta0.sort_values(['nom_loc'])

In [61]:
consulta0.columns=['nom_loc','b12','Prom_Arriendo']

In [62]:
consulta0

nom_loc  b12  Prom_Arriendo
0   Antonio Nariño    1  505168.086207
1   Antonio Nariño    2  403799.983673
2   Antonio Nariño    3  196076.923077
3   Barrios Unidos    1  543703.694444
4   Barrios Unidos    2  519629.018433
..             ...  ...            ...
74            Usme    3  136315.789474
75            Usme    4  175000.000000
72            Usme    1  230141.666667
73            Usme    2  237740.256494
76            Usme    5    9999.000000

[77 rows x 3 columns]

# Punto 1:
Adicional al promedio calculado en el punto anterior, calcule el mínimo, el máximo, los cuartiles, la desviación estándar y el coeficiente de variación y realice el conteo de hogares. 

In [71]:
def q1(x):
    return x.quantile(0.25)

def q2(x):
    return x.quantile(0.75)

def cv(x):
    return np.std(x) / np.mean(x) * 100


resumen = {'c10_valor': ['count','min','mean', q1, 'median', q2,'std', 'max', cv]}

In [72]:
consulta1=punto0.query("c1 == 3").groupby(['nom_loc', 'b12'], as_index=False).agg(resumen)

In [73]:
consulta1

nom_loc b12 c10_valor                                               \
                           count       min           mean        q1    median   
0   Antonio Nariño   1       116    9999.0  505168.086207  300000.0  450000.0   
1   Antonio Nariño   2       245    9999.0  403799.983673  300000.0  400000.0   
2   Antonio Nariño   3        39  100000.0  196076.923077  150000.0  180000.0   
3   Barrios Unidos   1       108    9999.0  543703.694444  250000.0  400000.0   
4   Barrios Unidos   2       217    9999.0  519629.018433  341000.0  450000.0   
..             ...  ..       ...       ...            ...       ...       ...   
72            Usme   1       120   22000.0  230141.666667  150000.0  230000.0   
73            Usme   2       308    9999.0  237740.256494  180000.0  250000.0   
74            Usme   3        38   20000.0  136315.789474  100000.0  132500.0   
75            Usme   4        12  100000.0  175000.000000  117500.0  200000.0   
76            Usme   5         1    9999.0    9999.000000    9999.0    9999.0   

                                                   
          q2            std        max         cv  
0   600000.0  314037.590559  2500000.0  61.896438  
1   500000.0  147790.313776   840000.0  36.525111  
2   220000.0   88148.912468   600000.0  44.376186  
3   700000.0  398792.307525  2000000.0  73.006999  
4   650000.0  288606.040311  2000000.0  55.412664  
..       ...            ...        ...        ...  
72  280000.0  108197.088750  1000000.0  46.816957  
73  300000.0  128708.413179  2000000.0  54.050291  
74  150000.0   44443.772715   250000.0  32.171686  
75  200000.0   60226.843909   300000.0  32.950179  
76    9999.0            NaN     9999.0   0.000000  

[77 rows x 11 columns]

# Punto 2: 
Basados en los cálculos anteriores, en qué localidad y en qué tipo de vivienda se encuentran los arriendos más económicas, en qué localidad y en qué tipo de vivienda se encuentran los valores más diversos en cuanto el precio del arriendo.

In [80]:
consulta2=punto0.query("c1 == 3").groupby(['nom_loc', 'b12'], as_index=False).agg({'c10_valor':['mean']})

In [82]:
consulta2.columns=['nom_loc','b12','prom_arriendo']
consulta2=consulta2.sort_values('prom_arriendo')
consulta2

nom_loc  b12  prom_arriendo
76           Usme    5   9.999000e+03
47   Rafael Uribe    5   7.000000e+04
33        Kennedy    3   1.000000e+05
67     Tunjuelito    4   1.100000e+05
51  San Cristobal    4   1.104545e+05
..            ...  ...            ...
68       Usaaquén    1   7.353414e+05
61    Teusaquillo    2   7.443358e+05
60    Teusaquillo    1   8.634583e+05
69       Usaaquén    2   8.809823e+05
17      Chapinero    2   1.161299e+06

[77 rows x 3 columns]

La clase 5 (otro tipo de vivienda) que es vivienda no formal y ubicados en la localidad de Usme

Punto 3
Usted es un empresario exitoso del sector inmobiliario de Bogotá, y quiere realizar una consulta del precio promedio de los arriendos en los estratos 5 y 6 en las localidades de Usaquén y Chapinero en donde usted desarrolla su negocio. Realice una única consulta para responder a su pregunta.

In [84]:
punto0.nom_loc.value_counts()

Bosa              1116
Tunjuelito        1057
Rafael Uribe      1032
Usme              1023
Puente Aranda      944
San Cristobal      941
Ciudad Bolivar     876
Barrios Unidos     862
Suba               857
Santa Fe           835
Engativa           833
Antonio Nariño     829
Kennedy            828
Teusaquillo        789
Usaaquén           774
Chapinero          744
Fontibón           738
Los Mártires       721
Candelaria         709
Name: nom_loc, dtype: int64

In [85]:
consulta3=punto0.query("(estrato == 5 | estrato == 6) & nom_loc in ['Usaquén','Chapinero']").c10_valor.mean()

In [86]:
consulta3

1685844.91

El promedio del ingreso es de 1685844.91


# 4.	
Calcule el promedio de cuartos por persona para cada localidad. Presente la tabla ordenada por localidad. 

In [130]:
punto4_c = c.filter(['directorio','c18','c20_cuartos',])
punto4=punto4_c.merge(a, on='directorio', how='inner')
punto4=punto4.merge(localidad, on='localidad', how='inner')

In [131]:
punto4.groupby(['nom_loc'], as_index=False).agg({'c20_cuartos':'mean'}).sort_values(['c20_cuartos'],ascending=False)


nom_loc  c20_cuartos
6         Engativa     2.350540
0   Antonio Nariño     2.295537
7         Fontibón     2.289973
10   Puente Aranda     2.275424
14            Suba     2.263711
17        Usaaquén     2.253230
8          Kennedy     2.243961
1   Barrios Unidos     2.178654
9     Los Mártires     2.170596
11    Rafael Uribe     2.156977
16      Tunjuelito     2.102176
12   San Cristobal     2.091392
15     Teusaquillo     2.091255
2             Bosa     2.065412
5   Ciudad Bolivar     2.026256
18            Usme     1.980450
13        Santa Fe     1.856287
4        Chapinero     1.795699
3       Candelaria     1.706629

# 5.	
¿Cuál es el hogar con mayor hacinamiento en Bogotá? (Calcule el cociente entre el número de personas sobre el número de cuartos). Cuál es el directorio de ese hogar. (DIRECTORIO_HOG).

In [133]:
punto4=punto4.assign(hacinamiento = lambda x: x.c18 / x.c20_cuartos).sort_values('hacinamiento', ascending=False)
punto4.head()

directorio  c18  c20_cuartos  localidad  estrato  sector  seccion  \
9032      219991   10            1         19        1    2519        5   
9245      220233    9            1         19        1    2525        2   
6586      217384    9            1          4        2    1308        1   
8994      219957    8            1         19        1    2561        4   
9047      220003    8            1         19        1    2562        1   

      manzana         nom_loc  hacinamiento  
9032       12  Ciudad Bolivar          10.0  
9245       15  Ciudad Bolivar           9.0  
6586       19   San Cristobal           9.0  
8994        2  Ciudad Bolivar           8.0  
9047       30  Ciudad Bolivar           8.0

# 6. En qué tipo de construcción se presentan las mayores tasas promedio de hacinamiento?

In [136]:
punto6_b = b.filter(['directorio','b12'])
punto6_c = c.filter(['directorio','c18','c20_cuartos',])
punto6=punto6_b.merge(punto4_c, on='directorio', how='inner')
punto6=punto6.merge(a, on='directorio', how='inner')
punto6=punto6.merge(localidad, on='localidad', how='inner')
punto6=punto6.assign(hacinamiento = lambda x: x.c18 / x.c20_cuartos).sort_values('hacinamiento', ascending=False)
punto6.head()

directorio  b12  c18  c20_cuartos  localidad  estrato  sector  seccion  \
9032      219991    1   10            1         19        1    2519        5   
6586      217384    3    9            1          4        2    1308        1   
9245      220233    2    9            1         19        1    2525        2   
9047      220003    5    8            1         19        1    2562        1   
5913      229085    1    8            1          2        1    8209        3   

      manzana         nom_loc  hacinamiento  
9032       12  Ciudad Bolivar          10.0  
6586       19   San Cristobal           9.0  
9245       15  Ciudad Bolivar           9.0  
9047       30  Ciudad Bolivar           8.0  
5913       30       Chapinero           8.0

In [138]:
punto6.groupby(['b12'], as_index=False).agg({'hacinamiento':'mean'}).sort_values(['hacinamiento'],ascending=False)


b12  hacinamiento
4    5      2.656863
2    3      2.053521
3    4      1.804245
0    1      1.672112
1    2      1.610772

Las casas con Otra forma de tenencia (posesión sin título, ocupante de hecho,
Pase a 9 propiedad colectiva, etc.) son las que más hacinamiento presentan.

# 7 	¿Cuál tipo de vivienda (B12) que presenta los mayores promedios en las tasas de hacinamiento en las localidades de Ciudad Bolívar y Usme?

In [139]:
punto7=punto6.query("nom_loc in ['Ciudad Bolivar','Usme']").groupby(['b12'], as_index=False).agg({'hacinamiento':'mean'}).sort_values(['hacinamiento'],ascending=False)
punto7

b12  hacinamiento
4    5      3.150000
2    3      2.169399
3    4      2.100000
1    2      2.035105
0    1      1.935619

Similarmente, las casas con Otra forma de tenencia (posesión sin título, ocupante de hecho,
Pase a 9 propiedad colectiva, etc.) son las que más hacinamiento presentan.

In [140]:
# 8.	Calcule el total de personas de la encuesta por localidad; total de personas de la encuesta por localidad y tipo de vivienda